include/extractor/turn_instructions.hpp

```diff
@@ -31,37 +36,98 @@ enum class TurnInstruction : unsigned char
     AccessRestrictionPenalty = 129
 };
 
-// Translates between angles and their human-friendly directional representation
-inline TurnInstruction getTurnDirection(const double angle)
+inline TurnInstruction shiftTurnToLeft(TurnInstruction turn)
 {
-    if (angle >= 23 && angle < 67)
-    {
-        return TurnInstruction::TurnSharpRight;
-    }
-    if (angle >= 67 && angle < 113)
+    switch (turn)
     {
+    case TurnInstruction::NoTurn:
+        return TurnInstruction::TurnSlightLeft;
+    case TurnInstruction::GoStraight:
+        return TurnInstruction::TurnSlightLeft;
+    case TurnInstruction::TurnSlightRight:
+        return TurnInstruction::GoStraight;
+    case TurnInstruction::TurnRight:
+        return TurnInstruction::TurnSlightRight;
+    case TurnInstruction::TurnSharpRight:
         return TurnInstruction::TurnRight;
+    case TurnInstruction::TurnLeft:
+        return TurnInstruction::TurnSharpLeft;
+    case TurnInstruction::TurnSlightLeft:
+        return TurnInstruction::TurnLeft;
+    default:
+        return turn;
     }
-    if (angle >= 113 && angle < 158)
+}
+
+inline TurnInstruction shiftTurnToRight(TurnInstruction turn)
+{
+    switch (turn)
     {
+    case TurnInstruction::NoTurn:
         return TurnInstruction::TurnSlightRight;
-    }
-    if (angle >= 158 && angle < 202)
-    {
+    case TurnInstruction::GoStraight:
+        return TurnInstruction::TurnSlightRight;
+    case TurnInstruction::TurnSlightRight:
+        return TurnInstruction::TurnRight;
+    case TurnInstruction::TurnRight:
+        return TurnInstruction::TurnSharpRight;
+    case TurnInstruction::TurnSharpLeft:
+        return TurnInstruction::TurnLeft;
+    case TurnInstruction::TurnLeft:
+        return TurnInstruction::TurnSlightLeft;
+    case TurnInstruction::TurnSlightLeft:
         return TurnInstruction::GoStraight;
+    default:
+        return turn;
     }
-    if (angle >= 202 && angle < 248)
-    {
+}
+
+inline double angularDeviation( const double angle, const double from ){
+  const double deviation = fabs( angle - from );
+  return std::min( 360 - deviation, deviation );
+}
+
+inline double getAngularPenalty(const double angle, TurnInstruction instruction)
+{
+    const double center[] = {180, 180, 135, 90, 45,
+                             0,   315, 270, 225}; // centers of turns from getTurnDirection
+    return angularDeviation( center[static_cast<int>(instruction)], angle);
+}
+
+inline double getTurnConfidence(const double angle, TurnInstruction instruction)
+{
+    // special handling of U-Turns and Roundabout
+    if (instruction >= TurnInstruction::HeadOn || instruction == TurnInstruction::UTurn ||
+        instruction == TurnInstruction::NoTurn)
+        return 1.0;
```

Comment:
Oh? It surprises me that the `>=` on instances of enum classes works?! Never knew that. Do you really want to make this dependant on the instruction order in the enum? I can see things breaking when we add or remove instructions.